internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -202,21 +234,31 @@ func (h *Diagnostics) diagFile(aDiag []client.DiagnosticFileResult, uDiag []clie
 			h.log.Warn(str)
 		}
 	}()
+
 	if err := diagnostics.ZipArchive(&wBuf, f, aDiag, uDiag); err != nil {
-		os.Remove(name)
+		rErr := os.Remove(name)
+		if rErr != nil {
+			h.log.Warnf("failed to create diagnostics zip, then failed to remove the file: %v", rErr)
```

Comment:
do you think it would be useful to have a reason for why this was actually entered? for these 3 error cases (logging `err` as well)